analyzer/codechecker_analyzer/analyzers/clangsa/analyzer.py

```diff
@@ -288,6 +290,37 @@ def get_analyzer_config(cls) -> List[str]:
 
         return parse_clang_help_page(command, 'OPTIONS:')
 
+    def post_analyze(self, result_handler):
+        """
+        Disabled checkers are not actually disabled during analysis, because
+        they rely on each other under the hood. The disabled checkers' reports
+        are removed in this post-processing step.
+        """
+        if not os.path.isfile(result_handler.analyzer_result_file):
+            return
+
+        try:
+            with open(result_handler.analyzer_result_file, 'rb') as f:
+                plist = plistlib.load(f)
+        except plistlib.InvalidFileException:
+            # It may happen that a compilation database contains a build action
+            # multiple times. For further details see analyzer_action_str() in
+            # ResultHandler. If this happens then the analysis of the same
+            # build action is analyzed in parallel several times, so the same
+            # output .plist file is changed by several threads. This may result
+            # an invalid .plist which fails plistlib.load(). This is not a big
+            # problem, because the second thread will also execute this
+            # post-processing and it happens rarely anyways.
+            # test_compile_uniqueing() fails undeterministically without this.
+            return
+
+        plist['diagnostics'] = list(filter(
```

Comment:
Plist manipulation can be painfully slow. How hard does this trickery hit the performance?

Reply:
I had this concern too, but this post-processing is applied only for ClangSA currently where we assume that this will be feasible (there are no verbose style checkers). For clang-tidy and cppcheck we don't even write these reports to the .plist. Nevertheless, we plan to apply this uniform post-processing method for those too, so it will be a relevant question later. Anyways, I'm not aware of a better solution with ClangSA, because it natively generates the .plst files.